Clarify `hours_to_expiration` for incremental models (#2837)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/bigquery-configs.md

By default, dbt-created tables never expire. You can configure certain model(s)
to expire after a set number of hours by setting `hours_to_expiration`.

:::info Note

> The `hours_to_expiration` only applies to initial creation of the underlying table. For incremental models, the `hours_to_expiration` is **not** reset on incremental runs.

Reviewer: great addition, Doug! added a non blocking suggestion @dbeatty10 to simplify the language. 

```suggestion
The `hours_to_expiration` only applies to initial creation of the underlying table. It doesn't reset for incremental models when they do another run.
```

Author: Love it, thank you @mirnawong1 🤩
